### DISCUSSION TITLE
Hello, how can you do multilanguage using ink, to have dialogues in English, Spanish and French?

### DISCUSSION THREAD

**Discussion by sakunix** [inkle/ink#772] [General]
Hello, how can you do multilanguage using ink, to have dialogues in English, Spanish and French?

thanks

**Comment by Selsynn** [comment]
unfortunately, there is no really great framework to do multilanguage in ink. 

One of the solution is to have different logic +text resource and cascade any changes. It's hard on the maintenance side, as any bug needs to be fixed in all language each time, but it's the only solution I know that let you use all the possibilities of the ink language itself. 

One of the other solution is to use the text itself of being shown to player directly, it's keys to be translated. The main drawback from that solution is that if you are a lot of change in text where one `line` of content can get out as a lot of different line, you need to keep them separated. Another drawback is that you delocalize you text and your logic, and that's one of the main feature I, personally, love about ink. 

It depends of the way your ink is structured. If it is logic light but you use a lot of feature of ink, I personally would recommend the first solution, and treating it as 3 or 4 different games. (especially if that's language with a lot of different grammar or sentence construction)
If you are not using a lot of change in the line but having a lot more heavy logic in it, I would go with the tag / or the id instead.
